src/metrics/mongoose_metrics_mam_hooks.erl

```diff
@@ -96,78 +97,77 @@ mam_lookup_messages(Result = {ok, {_TotalCount, _Offset, MessageRows}},
     {ok, Acc}.
 
 %% ----------------------------------------------------------------------------
 %% mod_mam_muc
 
 -spec mam_muc_get_prefs(Acc, Params, Extra) -> {ok, Acc} when
-    Acc :: any(),
+    Acc :: mod_mam:preference() | {error, Reason :: term()},
     Params :: map(),
     Extra :: gen_hook:extra().
-mam_muc_get_prefs(Acc, _, #{host_type := HostType}) ->
+mam_muc_get_prefs(Result, _Params, #{host_type := HostType}) ->
     mongoose_metrics:update(HostType, modMucMamPrefsGets, 1),
-    {ok, Acc}.
+    {ok, Result}.
 
 -spec mam_muc_set_prefs(Acc, Params, Extra) -> {ok, Acc} when
-    Acc :: any(),
+    Acc :: term(),
     Params :: map(),
     Extra :: gen_hook:extra().
-mam_muc_set_prefs(Acc, _, #{host_type := HostType}) ->
+mam_muc_set_prefs(Result, _Params, #{host_type := HostType}) ->
     mongoose_metrics:update(HostType, modMucMamPrefsSets, 1),
-    {ok, Acc}.
+    {ok, Result}.
 
 -spec mam_muc_remove_archive(Acc, Params, Extra) -> {ok, Acc} when
-    Acc :: any(),
-    Params :: map(),
+    Acc :: term(),
+    Params :: #{archive_id := mod_mam:archive_id() | undefined, room := jid:jid()},
     Extra :: gen_hook:extra().
-mam_muc_remove_archive(Acc, _, #{host_type := HostType}) ->
+mam_muc_remove_archive(Acc, _Params, #{host_type := HostType}) ->
     mongoose_metrics:update(HostType, modMucMamArchiveRemoved, 1),
     {ok, Acc}.
 
--spec mam_muc_lookup_messages(Result, Params, Extra) -> {ok, Result} when
-    Result :: {ok | error, mod_mam:lookup_result()},
-    Params :: map(),
+-spec mam_muc_lookup_messages(Acc, Params, Extra) -> {ok, Acc} when
+    Acc :: {ok, mod_mam:lookup_result()} | {error, term()},
+    Params :: mam_iq:lookup_params(),
     Extra :: gen_hook:extra().
-mam_muc_lookup_messages(Result = {ok, {_TotalCount, _Offset, MessageRows}},
-                        _, #{host_type := HostType}) ->
+mam_muc_lookup_messages({ok, {_TotalCount, _Offset, MessageRows}} = Result,
+                        _Params,
+                        #{host_type := HostType}) ->
     mongoose_metrics:update(HostType, modMucMamForwarded, length(MessageRows)),
     mongoose_metrics:update(HostType, modMucMamLookups, 1),
     {ok, Result};
-mam_muc_lookup_messages(Result = {error, _}, _, _) ->
+mam_muc_lookup_messages(Result = {error, _}, _Params, _Extra) ->
     {ok, Result}.
 
 -spec mam_muc_archive_message(Acc, Params, Extra) -> {ok, Acc} when
-    Acc :: any(),
+    Acc :: ok | {error, timeout},
     Params :: mod_mam:archive_message_params(),
     Extra :: gen_hook:extra().
-mam_muc_archive_message(Acc, _, #{host_type := HostType}) ->
+mam_muc_archive_message(Result, _Params, #{host_type := HostType}) ->
     mongoose_metrics:update(HostType, modMucMamArchived, 1),
-    {ok, Acc}.
+    {ok, Result}.
 
 -spec mam_muc_flush_messages(Acc, Params, Extra) -> {ok, Acc} when
-    Acc :: any(),
-    Params :: #{message_count := integer()},
+    Acc :: ok,
+    Params :: map(),
```

Comment:
Also here it is also `count := integer()`, no losing the type information better